### PR TITLE
Ipv6 support

### PR DESCRIPTION
**Pull request by bentsi**
## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I gave variables/functions meaningful self-explanatory names
- [ ] I didn't leave commented-out/debugging code
- [ ] I didn't copy-paste code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/logcollector.py [hunk 1]
```diff
@@ -318,7 +318,7 @@ def __init__(self, *args, **kwargs):
             test_start_time = time.time() - (6 * 3600)
         self.start_time = str(test_start_time).split('.')[0] + '000'
         super(GrafanaEntity, self).__init__(*args, **kwargs)
-        self.install_phantom_js()
+        # self.install_phantom_js()
```
Comment (fruch) [inline sdcm/logcollector.py]: why this is commented out ?

File: sdcm/cluster.py [hunk 2]
```diff
@@ -1384,11 +1386,11 @@ def config_setup(self, seed_address=None, cluster_name=None, enable_exp=True, en
 
             # Set listen_address
             pattern = re.compile('listen_address:.*')
-            scylla_yaml_contents = pattern.sub('listen_address: {0}'.format(self.private_ip_address),
```
Comment (roydahan) [inline sdcm/cluster.py]: I'm not sure it's correct.

File: sdcm/cluster.py [hunk 3]
```diff
@@ -1384,11 +1386,11 @@ def config_setup(self, seed_address=None, cluster_name=None, enable_exp=True, en
...
+            scylla_yaml_contents = pattern.sub('listen_address: {0}'.format(self.ip_address),
                                                scylla_yaml_contents)
             # Set rpc_address
             pattern = re.compile('\n[# ]*rpc_address:.*')
-            scylla_yaml_contents = pattern.sub('\nrpc_address: {0}'.format(self.private_ip_address),
```
Comment (roydahan) [inline sdcm/cluster.py]: Same here.

File: sdcm/cluster.py [hunk 4]
```diff
@@ -3317,7 +3319,7 @@ def node_run_stress_bench(node, loader_idx, stress_cmd, node_list):
             log_file_name = os.path.join(logdir,
                                          'scylla-bench-l%s-%s.log' %
                                          (loader_idx, uuid.uuid4()))
-            ips = ",".join([n.private_ip_address for n in node_list])
```
Comment (roydahan) [inline sdcm/cluster.py]: Need to test it scylla_bench using public and private IP

File: sdcm/tester.py [hunk 5]
```diff
@@ -271,7 +271,7 @@ def setUp(self):
             self.log.info('repair system_auth keyspace ...')
             node.run_nodetool(sub_cmd="repair", args="-- system_auth")
 
-        db_node_address = self.db_cluster.nodes[0].private_ip_address
```
Comment (roydahan) [inline sdcm/tester.py]: How did it work till now when we used public ip?

File: sdcm/tester.py [hunk 6]
```diff
@@ -1581,8 +1581,8 @@ def get_data_set_size(self, cs_cmd):
     @retrying(n=60, sleep_time=60, allowed_exceptions=(AssertionError,))
     def wait_data_dir_reaching(self, size, node):
         query = '(sum(node_filesystem_size{{mountpoint="{0.scylla_dir}", ' \
-            'instance=~"{1.private_ip_address}"}})-sum(node_filesystem_avail{{mountpoint="{0.scylla_dir}", ' \
-            'instance=~"{1.private_ip_address}"}}))'.format(self, node)
+            'instance=~"{1.external_address}"}})-sum(node_filesystem_avail{{mountpoint="{0.scylla_dir}", ' \
+            'instance=~"{1.external_address}"}}))'.format(self, node)
```
Comment (roydahan) [inline sdcm/tester.py]: why external and not ip_address?
How did it work till now?

File: test-cases/longevity/longevity-10gb-3h.yaml [hunk 7]
```diff
@@ -13,3 +13,9 @@ nemesis_interval: 5
 
 user_prefix: 'longevity-10gb-3h'
 space_node_threshold: 64424
+ip_ssh_connections: "ipv6"
+
+
+post_behavior_db_nodes: destroy
+post_behavior_loader_nodes: destroy
+post_behavior_monitor_nodes: destroy
```
Comment (roydahan) [inline test-cases/longevity/longevity-10gb-3h.yaml]: need to remove these

File: test-cases/longevity/longevity-10gb-3h.yaml [hunk 8]
```diff
@@ -13,3 +13,9 @@ nemesis_interval: 5
 
 user_prefix: 'longevity-10gb-3h'
 space_node_threshold: 64424
+ip_ssh_connections: "ipv6"
```
Comment (roydahan) [inline test-cases/longevity/longevity-10gb-3h.yaml]: Not sure we want it in this longevity specifically, and if so we may want to add it in the name.

**Comment by roydahan** [comment]
In addition to these changes, in order for IPv6 to work we need to add in scylla.yaml the following parameter:

'enable_ipv6_dns_lookup: true'

This may be changed if we will support the params that cassandra supports.
prefer_ipv6 params.

**Pull request closed without merging**